cmd/kogito/command/install/data_index.go

```diff
@@ -152,7 +151,6 @@ func (i *installDataIndexCommand) InitHook() {
 }
 
 func (i *installDataIndexCommand) Exec(cmd *cobra.Command, args []string) error {
-	log := context.GetDefaultLogger()
 	var err error
 	if i.flags.Project, err = shared.EnsureProject(i.Client, i.flags.Project); err != nil {
 		return err
```

Comment:
Just found a bug on line#165
`if available, err := infrastructure.IsInfinispanOperatorAvailable(i.Client, i.flags.Project); err != nil `

when operator was not installed, then it does install it. but the dependent infinispan operator is not yet installed and this line will return `not available` and we get an error when trying directly to install the data-index...

```
$ oc new-project tradisso-test
$ kogito install data-index
infinispan-url not informed, Infinispan will be automatically deployed via Infinispan Operator
No Kafka information has been given. A Kafka instance will be automatically deployed via Strimzi Operator in the namespace. Kafka Topics will be created accordingly if they don't exist already
Error: Infinispan Operator is not available in the Project: tradisso-test. Please make sure to install it before deploying Data Index without infinispan-url provided 
```

Reply:
I'm afraid we can't do much about it at this time. On OpenShift I can look for the subscription on Infinispan, but on Kubernetes I won't be able to look at that.

I think we can let the CLI to install the Data Index (and the Operator now that we have the `ServiceInstallationBuilder`), and only blocks it if the CRD is not present.

Then we can emit a WARN to users that the Infinispan Types are there, but the operator might not be ready yet (since we haven't found any deployments of the given operator).

I'll update the PR to reflect this.

wdyt?

Reply:
PR updated. Take a look and tell me wdyt:

1. I'll keep the `OperatorInstalled` flag to not have to roundtrip again and leave the operator to be installed first 
2. The Infinispan / Kafka check now just logs in the console that the operators might not be ready, but continue the installation any way. Since we do not have a "hard" dependency, since the operator will handle that for us by enforcing the operators to be there, we do not annoy the user
3. This code will be moved to the framework as soon as we agreed with that in #318 PR

Reply:
good for me. Check my other comment on `use_project` and `new_project`